kotlin/codegen/src/test/java/com/squareup/moshi/kotlin/codegen/JsonClassCodegenProcessorTest.kt

```diff
@@ -355,6 +355,27 @@ class JsonClassCodegenProcessorTest {
       "Invalid option value for ${JsonClassCodegenProcessor.OPTION_GENERATED}"
     )
   }
+  @Test
+  fun disableProguardGenerating() {
+    val result = prepareCompilation(
+      kotlin(
+        "source.kt",
+        """
+          import com.squareup.moshi.JsonClass
+
+          @JsonClass(generateAdapter = true)
+          data class Foo(val a: Int)
+          """
+      )
+    ).apply {
+      kaptArgs[JsonClassCodegenProcessor.OPTION_PROGUARD_CODE_GENERATED] = "false"
+    }.compile()
+    assertThat(result.messages).contains(
```

Comment:
After removing the warning, change this test to just assert that no rules files were generated

Reply:
by removing, I replaced this with result.generatedFiles assertion. bf273da